Improve multi tenancy documentation (#2620)
giantswarm/docs: src/content/tutorials/observability/multi-tenancy/creating-grafana-organization/_index.md

---
linkTitle: Creating a Grafana organization
title: Creating a Grafana organization
description: Step-by-step guide to create and configure Grafana organizations for multi-tenant observability.
menu:
  principal:
    identifier: tutorials-observability-multitenancy-create-grafana-organization
    parent: tutorials-observability-multitenancy
weight: 40
last_review_date: 2025-06-11
user_questions:
  - How to create a Grafana organization?
  - How to configure RBAC for Grafana organizations?
  - How to assign tenants to organizations?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

This guide walks you through creating and configuring Grafana organizations to implement [multi-tenancy]({{< relref "/tutorials/observability/multi-tenancy" >}}) in your observability platform.

## Prerequisites

Before creating your organization, make sure you have:

> - Access to the management cluster where you can create custom resources

Reviewer: ```suggestion

- Access to the management cluster, where you can create custom resources
```

---
linkTitle: Creating a Grafana organization
title: Creating a Grafana organization
description: Step-by-step guide to create and configure Grafana organizations for multi-tenant observability.
menu:
  principal:
    identifier: tutorials-observability-multitenancy-create-grafana-organization
    parent: tutorials-observability-multitenancy
weight: 40
last_review_date: 2025-06-11
user_questions:
  - How to create a Grafana organization?
  - How to configure RBAC for Grafana organizations?
  - How to assign tenants to organizations?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

> This guide walks you through creating and configuring Grafana organizations to implement [multi-tenancy]({{< relref "/tutorials/observability/multi-tenancy" >}}) in your observability platform.

Reviewer 2: It is also worth mentioning that a GrafanaOrganization is what dictate the validity of a tenant.

Author: ```suggestion
This guide walks you through creating and configuring Grafana organizations to implement [multi-tenancy]({{< relref "/tutorials/observability/multi-tenancy" >}}) in your observability platform. We strongly advise that you read this documentation first.
```

Reviewer 2: :+1:

Reviewer 2: :+1:

## Prerequisites

Before creating your organization, make sure you have:

- Access to the management cluster, where you can create custom resources
- Identified the [tenant names]({{< relref "/tutorials/observability/multi-tenancy#tenant-naming-best-practices" >}}) you want to use

> - Configured identity provider groups for RBAC

Reviewer 3: Is there a doc for that?

Author: not really, this will most likely be for shield I think

### Group format

Most Giant Swarm installations use Dex as the identity provider. Specify groups using the format `{dex-connector-id}:{group-name}`:

```yaml
rbac:
  admins:
  - customer:platform-admin        # Maps 'platform-admin' group from 'customer' connector
  - customer:ops-team
  editors:
  - customer:development-team
  viewers:
  - customer:support-team
```

**Finding your connector ID:** Check your cluster's Dex configuration for the connector ID (usually `customer`).

**Required fields:** Only `admins` is required; `editors` and `viewers` are optional.

## What happens when you create an organization

Creating a `GrafanaOrganization` resource automatically provisions:

1. **New Grafana organization** with your specified display name
2. **Tenant-scoped datasources** for Loki, Mimir, and Alertmanager
3. **User role assignments** based on your RBAC configuration
4. **Tenant validation** so only defined tenants can receive data (see [tenant governance]({{< relref "/tutorials/observability/multi-tenancy#tenant-governance" >}}))

## Verification steps

After creating your organization:

1. **Check organization status:**

>    ```bash

Reviewer: ```suggestion

   ```bash
```

| **Viewer** | Read-only access to dashboards and data |

### Group format

Most Giant Swarm installations use Dex as the identity provider. Specify groups using the format `{dex-connector-id}:{group-name}`:

```yaml
rbac:
  admins:
  - customer:platform-admin        # Maps 'platform-admin' group from 'customer' connector
  - customer:ops-team
  editors:
  - customer:development-team
  viewers:
  - customer:support-team
```

**Finding your connector ID:** Check your cluster's Dex configuration for the connector ID (usually `customer`).

**Required fields:** `admins` is mandatory; `editors` and `viewers` are optional.

## What happens when you create an organization

Creating a `GrafanaOrganization` resource automatically provisions:

1. **New Grafana organization** with your specified display name
2. **Tenant-scoped datasources** for Loki, Mimir, and Alertmanager
3. **User role assignments** based on your RBAC configuration
4. **Data collection** of alerts, metrics and logs

## Verification steps

After creating your organization:

1. **Check organization status:**

```bash
kubectl get grafanaorganization myonlineshop -o yaml
```

> 2. [**Log in to Grafana**]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}) and verify:

Author: ```suggestion
2. [**Log in to Grafana**]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}) and verify:

```

- The organization dropdown menu on the top-left corner shows all expected organizations

> ![Switching organization](./organization%20switching.png)

Author: ```suggestion
![Switching organization](./organization_switching.png)
```

| **Viewer** | Read-only access to dashboards and data |

### Group format

Most Giant Swarm installations use Dex as the identity provider. Specify groups using the format `{dex-connector-id}:{group-name}`:

```yaml
rbac:
  admins:
  - customer:platform-admin        # Maps 'platform-admin' group from 'customer' connector
  - customer:ops-team
  editors:
  - customer:development-team
  viewers:
  - customer:support-team
```

**Finding your connector ID:** Check your cluster's Dex configuration for the connector ID (usually `customer`).

**Required fields:** `admins` is mandatory; `editors` and `viewers` are optional.

## What happens when you create an organization

Creating a `GrafanaOrganization` resource automatically provisions:

1. **New Grafana organization** with your specified display name
2. **Tenant-scoped datasources** for Loki, Mimir, and Alertmanager
3. **User role assignments** based on your RBAC configuration
4. **Data collection** of alerts, metrics and logs

## Verification steps

After creating your organization:

1. **Check organization status:**

```bash
kubectl get grafanaorganization myonlineshop -o yaml

> ```

Author: ```suggestion
```

```

> 

Author: ```suggestion
```

### Dex connector format
| Role | Permissions |
|------|-------------|
| **Admin** | Full organization access: manage users, datasources, dashboards, and settings |
| **Editor** | Create and edit dashboards, alerts, and folders (cannot manage users) |
| **Viewer** | Read-only access to dashboards and data |

### Group format

Most Giant Swarm installations use Dex as the identity provider. Specify groups using the format `{dex-connector-id}:{group-name}`:

```yaml
rbac:
  admins:
  - customer:platform-admin        # Maps 'platform-admin' group from 'customer' connector
  - customer:ops-team
  editors:
  - customer:development-team
  viewers:
  - customer:support-team
```

Most Giant Swarm installations use Dex as the identity provider. When using Dex, specify groups using the format `{dex-connector-id}:{group-name}`. For example:
**Finding your connector ID:** Check your cluster's Dex configuration for the connector ID (usually `customer`).

- `customer:platform-admin` - Maps the `platform-admin` group from the `customer` Dex connector to Grafana Admin role

> - `customer:development-team` - Maps the `development-team` group from the `customer` Dex connector to Grafana Editor role

Reviewer 2: Please keep the section on dex connector groups mapping.

Is there a doc to link about dex ? To explain where to find/configure the dex connector and also where are groups configured.

Reviewer 2: Well if we have no doc we can't link it then :shrug:

Reviewer 2: Well if we have no doc we can't link it then :shrug:

> 4. **Tenant validation** so only defined tenants can receive data (see [tenant governance]({{< relref "/tutorials/observability/multi-tenancy#tenant-governance" >}}))

Author: This is about data collection and not visualization

> 4. **Tenant validation** so only defined tenants can receive data (see [tenant governance]({{< relref "/tutorials/observability/multi-tenancy#tenant-governance" >}}))

Author: ```suggestion
4. **Data collection** so telemetry collectors start collecting data from configured tenants
```

## Verification steps

After creating your organization:

1. **Check organization status:**

   ```bash
   kubectl get grafanaorganization myonlineshop -o yaml
   ```

2. **Log in to Grafana** and verify:

>    - New organization appears in the organization dropdown
>    - Datasources are configured for your tenants

Reviewer 3: wondering how to check "datasources are configured for your tenants".
I was going for a simple switch to active voice, but you can't verify anything like that:
```
The `Data sources" section shows connections for each of your tenants
```

So, so far my best suggestion is:
```suggestion
   - The organization dropdown menu shows all expected organization
   - You can explore and query data for each of your tenants
```

Reviewer 3: Oh, the next section describes how you "test data access". So I guess that's how you can explore you have data for your tenants, right? Or is there an easy step before?

>    - Datasources are configured for your tenants

Author: A link is hard because the url is installation dependant

   - Users have appropriate role assignments

3. **Test data access:**
   - Switch to your new organization in Grafana
   - Query data using the provisioned datasources

>    - Verify only your tenant's data is visible

Author: I'm open to ideas
